### ISSUE TITLE
const_oid: Add missing OIDs

### ISSUE THREAD

**Issue by bkstein** [RustCrypto/formats#688]
Some OIDs are missing. Particularly

- SHA3-224 algorithm: 2.16.840.1.101.3.4.2.7
- SHA3-256 algorithm: 2.16.840.1.101.3.4.2.8
- SHA3-384 algorithm: 2.16.840.1.101.3.4.2.9
- SHA3-512 algorithm: 2.16.840.1.101.3.4.2.10
- SHAKE128 algorithm: 2.16.840.1.101.3.4.2.11
- SHAKE256 algorithm: 2.16.840.1.101.3.4.2.12
- SM3 hash algorithm: 1.2.156.10197.1.401

I would like to contribute to formats/const_oid, but for some reason, the OID generation is well hidden. I don't know, how to add missing OIDs in crate const_oid.

**Comment by tarcieri** [comment]
The OID database is auto-generated from RFCs using this tool: https://github.com/RustCrypto/formats/tree/master/const-oid/oiddbgen

**Comment by bkstein** [comment]
Looks like SHA3 OIDs are not specified in any RFC. However, RFC 8692 references
```
   [SHA3]     National Institute of Standards and Technology, "SHA-3
              Standard: Permutation-Based Hash and Extendable-Output
              Functions", DOI 10.6028/NIST.FIPS.202, FIPS PUB 202,
              August 2015, <https://doi.org/10.6028/NIST.FIPS.202>.
```
which in chapter C references NIST:
```
C Object Identifiers
Object identifiers (OIDs) for SHA3-224, SHA3-256, SHA3-384, SHA3-512, SHAKE128, and
SHAKE256 are posted at http://csrc.nist.gov/groups/ST/crypto_apps_infra/csor/algorithms.html.
```

Possible solution would be to manually collect OIDs in a document, e.g. in github.com\RustCrypto\formats\const-oid\oiddbgen\non-rfc-oids.txt. We could use the same ASN.1 specification syntax and feed the parser. Does that sound reasonable?

**Comment by tarcieri** [comment]
It'd be nice to name them after the document which defines them, e.g. `fips202.txt`.

Alternatively you could add them to the `sha3` crate.

**Comment by bkstein** [comment]
This was implemented with anniversary PR #1000.
